src/components/NcAppNavigationSearch/NcAppNavigationSearch.vue

```diff
@@ -0,0 +1,272 @@
+}
+
+.styleguide-wrapper {
+	background-color: var(--color-background-plain);
+	padding: var(--body-container-margin);
+}
+</style>
+```
+</docs>
+<template>
+	<div class="app-navigation-search"
+		:class="{
+			'app-navigation-search--has-actions': hasActions(),
+		}">
+		<NcInputField ref="inputElement"
+			:aria-label="label"
+			class="app-navigation-search__input"
+			label-outside
+			:placeholder="placeholder ?? label"
+			show-trailing-button
+			:trailing-button-label="t('Clear search')"
+			type="search"
+			:value="modelValue"
+			@update:value="$emit('update:modelValue', $event)"
+			@trailing-button-click="onCloseSearch">
+			<template #trailing-button-icon>
+				<IconClose :size="20" />
+			</template>
+		</NcInputField>
+		<div v-if="hasActions()"
+			ref="actionsContainer"
+			class="app-navigation-search__actions"
+			:class="{
+				'app-navigation-search__actions--hidden': !showActions,
+				'hidden-visually': hideActions,
+			}">
+			<slot name="actions" />
+		</div>
+	</div>
+</template>
+
+<script setup>
+import { useFocusWithin } from '@vueuse/core'
+import { ref, nextTick, useSlots, watch } from 'vue'
+import { t } from '../../l10n.js'
+
+import IconClose from 'vue-material-design-icons/Close.vue'
+import NcInputField from '../NcInputField/NcInputField.vue'
+
+defineProps({
+	/**
+	 * Current search input
+	 */
+	modelValue: {
+		type: String,
+		default: '',
+	},
+	/**
+	 * Text used to label the search input
+	 */
+	label: {
+		type: String,
+		default: t('Search…'),
+	},
+
+	/**
+	 * Placeholder of the search input
+	 * By default the value of `label` is used.
+	 */
+	placeholder: {
+		type: String,
+		default: null,
+	},
+})
+
+const emit = defineEmits(['update:modelValue'])
+const slots = useSlots()
+
+const inputElement = ref()
+const { focused: inputHasFocus } = useFocusWithin(inputElement)
+
+/** Timeout used to define when the search input is fully expanded */
+const transitionTimeout = Number.parseInt(window.getComputedStyle(window.document.body).getPropertyValue('--animation-quick')) || 100
+
+/**
+ * @type {import('vue').Ref<import('vue').ComponentPublicInstance>}
+ */
+const actionsContainer = ref()
+const hasActions = () => !!slots.actions
+const showActions = ref(true)
+const timeoutId = ref()
+/** This is used to hide the actions after the transition is done */
+const hideActions = ref(false)
+watch(inputHasFocus, () => {
+	showActions.value = !inputHasFocus.value
+	window.clearTimeout(timeoutId.value)
+	if (showActions.value) {
+		hideActions.value = false
+	} else {
+		// If the transition is done, we fully hide the actions visually
+		window.setTimeout(() => {
+			hideActions.value = !showActions.value
+		}, transitionTimeout)
+	}
+})
+
+/**
+ * Handle close the search
+ */
+function onCloseSearch() {
+	emit('update:modelValue', '')
+	if (hasActions()) {
+		showActions.value = true
+		nextTick(() => actionsContainer.value.$el.querySelector('button')?.focus())
+	}
```

Comment:
This adds a new focus but doesn't prevent the previous one.

So, when you press <kbd>Tab</kbd>, focus first moves to real next element and only then after `100ms` makes a new move to actions. But it still counts as 2 independent moves for a web-browser.

Not sure it will be a problem in apps. In docs you can see it as a page scroll.

Reply:
But this is only called on button click, not on tab.